### PR TITLE
Add a function in PipelineDClient to call GetStats RPC endpoint

### PR DESCRIPTION
**Pull request by veshkemburu**
Signed-off-by: veshkemburu <veshkemburu@fb.com>

## Summary

Added rpc endpoint functions in PipelinedClient, and unit tests verifying functionality

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/session_manager/PipelinedClient.cpp [hunk 1]
```diff
@@ -423,6 +423,15 @@ void AsyncPipelinedClient::update_subscriber_quota_state(
   });
 }
 
+void AsyncPipelinedClient::poll_stats(cookie : int = 0, cookie_mask : int = 0){
+  auto req = GetStatsRequest(cookie = cookie, cookie_mask = cookie_mask);
```
Comment (koolzz) [inline lte/gateway/c/session_manager/PipelinedClient.cpp]: I think you're missing the GetStatsRequest func

File: lte/gateway/c/session_manager/PipelinedClient.cpp [hunk 2]
```diff
@@ -423,6 +431,15 @@ void AsyncPipelinedClient::update_subscriber_quota_state(
   });
 }
 
+void AsyncPipelinedClient::poll_stats(cookie : int = 0, cookie_mask : int = 0){
```
Comment (koolzz) [inline lte/gateway/c/session_manager/PipelinedClient.cpp]: remove the default argument values here

File: lte/gateway/c/session_manager/test/SessiondMocks.h [hunk 3]
```diff
@@ -52,6 +52,7 @@ class MockPipelined final : public Pipelined::Service {
         .WillByDefault(Return(Status::OK));
     ON_CALL(*this, SetupUEMacFlows(_, _, _)).WillByDefault(Return(Status::OK));
     ON_CALL(*this, SetupQuotaFlows(_, _, _)).WillByDefault(Return(Status::OK));
+    ON_CALL(*this, PollStats(_, _, _)).WillByDefault(Return(Status::OK));
```
Comment (koolzz) [inline lte/gateway/c/session_manager/test/SessiondMocks.h:55]: This should be GetStats ?

Comment (koolzz) [inline lte/gateway/c/session_manager/test/SessiondMocks.h]: This is mocking grpc so it should be GetStats instead of PollStats

**Review by koolzz — APPROVED** [review]

File: lte/gateway/c/session_manager/PipelinedClient.h [hunk 4]
```diff
@@ -288,6 +288,8 @@ class AsyncPipelinedClient : public GRPCReceiver, public PipelinedClient {
       const magma::UEMacFlowRequest req, const int retries, Status status,
       FlowResponse resp);
 
+  void poll_stats(int cookie, int cookie_mask);
```
Comment (themarwhal) [inline lte/gateway/c/session_manager/PipelinedClient.h]: add docs on what cookie and cookie mask are. (I have no idea what they are :) )
```
  /**
   * @brief TODO
   *
   * @param cookie // TODO
   * @param cookie_mask // TODO
   */
```

Comment (themarwhal) [reply lte/gateway/c/session_manager/PipelinedClient.h]: You could just go ahead and add `std::function<void(Status, RuleRecordTable)> callback` as a parameter for this function, as you'll need it anyways.

**Comment by themarwhal** [comment]
Can we remove `IN PROGRESS` from the PR title?

**Comment by themarwhal** [comment]
PR looks good. Please address the whitespace lints. 
And it's good practice to have the [commit message be in imperative mood](https://docs.magmacore.org/docs/contributing/contribute_workflow#guidelines).

So maybe modify the PR title to something like: `Add a function in PipelineDClient to call GetStats RPC endpoint`

**Review by themarwhal — APPROVED** [review]

**Pull request merged**